sqrl-engines/sqrl-engine-jdbc/src/main/java/com/datasqrl/engine/database/relational/JDBCEngine.java

```diff
@@ -161,16 +162,18 @@ private List<SqlDDLStatement> extractTypeExtensions(RelNode relNode) {
     Set<SqlDDLStatement> statements = new HashSet<>();
     //look at relnodes to see if we use a vector type
     for (RelDataTypeField field : relNode.getRowType().getFieldList()) {
-      if (field.getType() instanceof VectorType) {
+      if (field.getType() instanceof RawRelDataType &&
+        ((RawRelDataType) field.getType()).getRawType().getOriginatingClass() == FlinkVectorType.class)
```

Comment:
Can we please generalize this into the dialect? I.e. check in the dialect whether a particular type needs an extension?

Reply:
Yeah, this was mentioned in PR message. `Registering extensions should be service loaded in`. I can do it now and drop the vector dependency, or make a new ticket.